pages/date-range-picker/custom-control.page.tsx

```diff
@@ -1,33 +1,86 @@
 
-import { i18nStrings, isValid } from './common';
+import AppContext from '../app/app-context';
+import {
+  applyDisabledReason,
+  checkIfDisabled,
+  DateRangePickerDemoContext,
+  dateRangePickerDemoDefaults,
+  DisabledDate,
+  generateI18nStrings,
+  generatePlaceholder,
+  isValid,
+} from './common';
 
 export default function DatePickerScenario() {
+  const { urlParams, setUrlParams } = useContext(AppContext as DateRangePickerDemoContext);
   const [value, setValue] = useState<DateRangePickerProps['value']>(null);
 
+  const dateOnly = urlParams.dateOnly ?? dateRangePickerDemoDefaults.dateOnly;
+  const monthOnly = false;
+  const disabledDates =
+    (urlParams.disabledDates as DisabledDate) ?? (dateRangePickerDemoDefaults.disabledDates as DisabledDate);
+  const withDisabledReason = urlParams.withDisabledReason ?? dateRangePickerDemoDefaults.withDisabledReason;
+
   return (
     <Box padding="s">
       <h1>Date range picker with custom control</h1>
+      <SpaceBetween direction="horizontal" size="s">
+        <label>
+          Disabled dates{' '}
+          <select
+            value={disabledDates}
+            onChange={event =>
+              setUrlParams({
+                disabledDates: event.currentTarget.value as DisabledDate,
+              })
+            }
+          >
+            <option value="none">None (Default)</option>
+            <option value="all">All</option>
+            <option value="only-even">Only even</option>
+            <option value="middle-of-page">Middle of {monthOnly ? 'year' : 'month'}</option>
+            <option value="end-of-page">End of {monthOnly ? 'year' : 'month'}</option>
+            <option value="start-of-page">Start of {monthOnly ? 'year' : 'month'}</option>
+            <option value="overlapping-pages">Overlapping {monthOnly ? 'years' : 'months'}</option>
+          </select>
+        </label>
+        <Checkbox
+          checked={withDisabledReason}
+          onChange={({ detail }) => setUrlParams({ withDisabledReason: detail.checked })}
+        >
+          Disabled reasons
+        </Checkbox>
+        <Checkbox
+          disabled={monthOnly}
+          checked={dateOnly}
+          onChange={({ detail }) => setUrlParams({ dateOnly: detail.checked })}
+        >
+          Date-only
+        </Checkbox>
+      </SpaceBetween>
       <FormField label="Date Range Picker field">
         <DateRangePicker
           value={value}
           onChange={e => setValue(e.detail.value)}
           locale={enLocale.code}
-          i18nStrings={i18nStrings}
+          i18nStrings={generateI18nStrings(dateOnly, monthOnly)}
           relativeOptions={[]}
-          placeholder="Filter by a date and time range"
+          placeholder={generatePlaceholder(dateOnly, monthOnly)}
           isValidRange={isValid}
           rangeSelectorMode="absolute-only"
+          isDateEnabled={(date: Date) => checkIfDisabled(date, disabledDates, monthOnly)}
```

Comment:
This argument type is not needed because `isDateEnabled` is already typed on  its own